service/endpoint_validation_test.go

```diff
@@ -416,3 +465,397 @@ func TestInvalidAvailabilityStatuses(t *testing.T) {
+	newEndpoint.ID = 0 // set it as zero to avoid hitting
+	newEndpoint.Role = &role
+
+	err := endpointDao.Create(&newEndpoint)
+	if err != nil {
+		t.Errorf("unexpected error when inserting a new endpoint for the test: %s", err)
+	}
+
+	editRequest, sourceId := setUpEndpointEditRequest()
+	editRequest.Role = &role
+
+	err = ValidateEndpointEditRequest(endpointDao, sourceId, &editRequest)
+
+	want := "the role already exists for the given source"
+	got := err.Error()
+
+	if want != got {
+		t.Errorf(`unexpected error when modifying an endpoint's role to an already existing one. Want "%s", got "%s"`, want, got)
+	}
+}
+
+// TestEditSchemeGetsDefaulted tests if the scheme gets properly defaulted when an invalid or missing scheme is
+// provided. Tested on an "edit endpoint" payload.
+func TestEditSchemeGetsDefaulted(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	editRequest, sourceId := setUpEndpointEditRequest()
+
+	editRequest.Scheme = nil
+
+	err := ValidateEndpointEditRequest(endpointDao, sourceId, &editRequest)
+
+	if err != nil {
+		t.Errorf("unexpected error when validating a nil scheme for an endpoint edit: %s", err)
+	}
+
+	if *editRequest.Scheme != defaultScheme {
+		t.Errorf(`unexpected scheme returned when editing an endpoint and not providing an scheme. Want "%s", got "%s"`, defaultScheme, *editRequest.Scheme)
+	}
+
+	invalidScheme := "/invalid"
+	editRequest.Scheme = &invalidScheme
+
+	err = ValidateEndpointEditRequest(endpointDao, sourceId, &editRequest)
+
+	if err != nil {
+		t.Errorf("unexpected error when editing an endpoint and giving it an invalid scheme: %s", err)
+	}
+
+	if *editRequest.Scheme != defaultScheme {
+		t.Errorf(`unexpected scheme set when editing an endpoint with an invalid scheme value. Want "%s", got "%s"`, defaultScheme, *editRequest.Scheme)
+	}
+}
+
+// TestEditEmptyHost tests if no error is returned even when an empty host is given. Tested on an "edit endpoint"
+// payload.
+func TestEditEmptyHost(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	editRequest, sourceId := setUpEndpointEditRequest()
+	tmp := ""
+	editRequest.Host = &tmp
+
+	err := ValidateEndpointEditRequest(endpointDao, sourceId, &editRequest)
+	if err != nil {
+		t.Errorf("unexpected error when editing an endpoint and giving it an empty host: %s", err)
+	}
+}
+
+// TestEditHostFqdnTooLong tests if an error is returned when a host which is too long is given. Tested on an "edit
+// endpoint" payload.
+func TestEditHostFqdnTooLong(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	editRequest, sourceId := setUpEndpointEditRequest()
+
+	// The "longHostname" variable holds a 256 char hostname
+	tmp := `aaaaaaaaaabbbbbbbbbbccccccccccddddddddddeeeeeeeeee
+		aaaaaaaaaabbbbbbbbbbccccccccccddddddddddeeeeeeeeee
+		aaaaaaaaaabbbbbbbbbbccccccccccddddddddddeeeeeeeeee
+		aaaaaaaaaabbbbbbbbbbccccccccccddddddddddeeeeeeeeee
+		aaaaaaaaaabbbbbbbbbbccccccccccddddddddddeeeeeeeeee
+		aaaaaa
+	`
```

Comment:
you can use here Repeat() method too
`strings.Repeat("a", 256)`

Reply:
TIL! This is very helpful! Thanks!